packages/ui/src/contacts/ViewContacts.tsx

```diff
@@ -1,50 +1,54 @@
-import React, { FunctionComponent, useCallback, useContext, useEffect } from 'react';
+import React, {
+  FunctionComponent,
+  useCallback,
+  useContext,
+  useEffect,
+} from 'react';
 import { Contact } from '@centrifuge/gateway-lib/models/contact';
 import ContactList from './ContactList';
 import { Preloader } from '../components/Preloader';
 import { httpClient } from '../http-client';
-import { AppContext } from '../App';
+import { AuthContext } from '../auth/Auth';
 import { useMergeState } from '../hooks';
 import { PageError } from '../components/PageError';
+import { goToHomePage } from '../utils/goToHomePage';
 
-type State  = {
-  loading: boolean,
-  error: any,
-  contacts: Contact[]
-}
+type State = {
+  loading: boolean;
+  error: any;
+  contacts: Contact[];
+};
 
 const ViewContacts: FunctionComponent = () => {
-
-  const [
-    {
-      loading,
-      contacts,
-      error,
-    },
-    setState] = useMergeState<State>({
+  const [{ loading, contacts, error }, setState] = useMergeState<State>({
     loading: true,
     error: null,
     contacts: [],
   });
 
+  const { user, token } = useContext(AuthContext);
 
-  const { user } = useContext(AppContext);
-
+  const displayPageError = useCallback(
+    error => {
+      setState({
+        loading: false,
+        error,
+        contacts: [],
+      });
+    },
+    [setState],
+  );
 
-  const displayPageError = useCallback((error) => {
-    setState({
-      loading: false,
-      error,
-      contacts: [],
-    });
-  }, [setState]);
+  if (!token) {
+    goToHomePage();
+  }
 
   const createContact = async (contact: Contact) => {
     setState({
       loading: true,
     });
     try {
-      await httpClient.contacts.create(contact);
+      await httpClient.contacts.create(contact, token!);
```

Comment:
What is the `token!` doing here, specifically the `!`?

Reply:
I see [here](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator) in the TypeScript docs. It is a non-null assertion operator. 👍

Reply:
👍